### PR TITLE
Strip libmonosgen linked library

### PR DESCRIPTION
**Pull request by directhex**
It seems in all Mono .symbols.nupkg packages, instead of a stripped .dylib/so and companion .dwarf/.dbg, we've just been shipping unstripped libs (and shipping unstripped libs in the non-symbols packages too). This commit largely duplicates the functions from eng/native/functions.cmake into an MSbuild proj, as our entire install logic is already in MSbuild not CMake. Moving to the common CMake functions would be a much larger task.

This should reduce our install size, and better mirror the rest of dotnet's debug infra & behaviour.

### CODE REVIEW

**Comment by directhex** [comment]
I should fire off an official build, since this affects package content

**Review by steveisok — APPROVED** [review]

File: src/mono/mono.proj [hunk 1]
```diff
@@ -475,6 +475,37 @@
...
+      <_Objcopy Condition="'$(Platform)' == 'arm'">arm-linux-$(_LinuxAbi)eabi$(_LinuxFloatAbi)-$(_Objcopy)</_Objcopy>
+      <_Objcopy Condition="'$(Platform)' == 'arm64'">aarch64-linux-$(_LinuxAbi)-$(_Objcopy)</_Objcopy>
+      <_Objcopy Condition="'$(Platform)' == 'x64'">x86_64-linux-$(_LinuxAbi)-$(_Objcopy)</_Objcopy>
+      <_Objcopy Condition="'$(Platform)' == 'x86'">i686-linux-$(_LinuxAbi)-$(_Objcopy)</_Objcopy>
+      <_Objcopy Condition="'$(TargetsAndroid)' == 'true'">$(ANDROID_NDK_ROOT)/toolchains/llvm/prebuilt/$(MonoToolchainPrebuiltOS)/bin/$(_Objcopy)</_Objcopy>
+    </PropertyGroup>
+      <!-- test viability of objcopy command -->
+      <Exec Condition="'$(BuildMonoAOTCrossCompilerOnly)' != 'true' and $([MSBuild]::IsOSPlatform('Linux'))" Command="$(_Objcopy) -V" IgnoreStandardErrorWarningFormat="true" ContinueOnError="WarnAndContinue" IgnoreExitCode="true" EchoOff="true" ConsoleToMsBuild="true">
+        <Output TaskParameter="ExitCode" PropertyName="_ObjcopyFound"/>
+      </Exec>
```
Comment (akoeplinger) [inline src/mono/mono.proj:498]: nit: the indentation is off here

**Review by akoeplinger — APPROVED** [review]
This probably also fixes https://github.com/dotnet/runtime/issues/45013, though we should check we don't impact XA/XI.

**Comment by akoeplinger** [comment]
What's the status of this PR?

**Comment by directhex** [comment]
I think it's fine, just running into flakiness in unrelated places in testing

**Comment by akoeplinger** [comment]
Retriggered the lanes just to be sure :)

**Comment by lateralusX** [comment]
@directhex we would need to strip libmono-component-*.so when doing dynamic components build (currently only on Android), something we could add into this PR as well?

**Comment by directhex** [comment]
@lateralusX how can i force a dynamic components build on... anything that isn't android? that's one platform i don't have the environment set up locally for

**Comment by directhex** [comment]
wait, docker, right. bleh

**Comment by directhex** [comment]
`Failed to create CoreCLR, HRESULT: 0x8007FF06`

I hate docker

**Comment by directhex** [comment]
@lateralusX this is looking good locally

```
$ unzip -lf /home/directhex/Projects/runtime/artifacts/packages/Release/Shipping/Microsoft.NETCore.App.Runtime.Mono.android-arm64.6.0.0-ci.symbols.nupkg | grep
dbg
  1717872  2021-06-08 14:05   runtimes/android-arm64/native/libmono-component-diagnostics_tracing.so.dbg
   475664  2021-06-08 14:05   runtimes/android-arm64/native/libmono-component-hot_reload.so.dbg
 17705560  2021-06-08 14:05   runtimes/android-arm64/native/libmonosgen-2.0.so.dbg
  2766008  2021-06-08 14:05   runtimes/android-arm64/native/libSystem.IO.Compression.Native.so.dbg
   313464  2021-06-08 14:05   runtimes/android-arm64/native/libSystem.Native.so.dbg
   751080  2021-06-08 14:05   runtimes/android-arm64/native/libSystem.Security.Cryptography.Native.Android.so.dbg

$ unzip -lf /home/directhex/Projects/runtime/artifacts/packages/Release/Shipping/Microsoft.NETCore.App.Runtime.Mono.android-arm64.6.0.0-ci.nupkg | grep dbg
```

**Comment by lateralusX** [comment]
> @lateralusX how can i force a dynamic components build on... anything that isn't android? that's one platform i don't have the environment set up locally for

Sorry, just saw this, there is `MonoComponentsStatic` in `mono.proj` controling build, so not setting it or make sure cmake won't get `-DSTATIC_COMPONENTS=1` should enable dynamic component builds.

**Comment by lateralusX** [comment]
What is the size of the stripped .so files?

**Comment by directhex** [comment]
@lateralusX 

```
-rw-r--r-- 3 root root 275K Jun  8 10:05 artifacts/bin/mono/Android.arm64.Release/libmono-component-diagnostics_tracing.so
-rwxr-xr-x 3 root root 1.7M Jun  8 10:05 artifacts/bin/mono/Android.arm64.Release/libmono-component-diagnostics_tracing.so.dbg
-rw-r--r-- 3 root root 131K Jun  8 10:05 artifacts/bin/mono/Android.arm64.Release/libmono-component-hot_reload.so
-rwxr-xr-x 3 root root 465K Jun  8 10:05 artifacts/bin/mono/Android.arm64.Release/libmono-component-hot_reload.so.dbg
-rw-r--r-- 3 root root 3.2M Jun  8 10:05 artifacts/bin/mono/Android.arm64.Release/libmonosgen-2.0.so
-rwxr-xr-x 3 root root  17M Jun  8 10:05 artifacts/bin/mono/Android.arm64.Release/libmonosgen-2.0.so.dbg
```

**Pull request merged**